harness/determined/profiler.py

```diff
@@ -768,21 +772,16 @@ def run(self) -> None:
                         f"be a bug in the code."
                     )
 
-            except queue.Empty:
-                pass
-
-            check.is_not_none(
-                batch_start_time,
-                "batch_start_time should never be None. The inbound_queue.get() "
-                "should never return and proceed to this piece of code "
-                "without batch_start_time being updated to a real timestamp. If "
-                "batch_start_time is None, inbound_queue.get() timeout should be "
-                "None and the get() should block until a Timing is received.",
-            )
-            batch_start_time = cast(float, batch_start_time)
-            if time.time() - batch_start_time > self.FLUSH_INTERVAL:
+            # Timeout met, or an explicit flush.
+            if not self.metrics_batch.isempty():
                 self.send_queue.put(self.metrics_batch.consume())
-                batch_start_time = time.time()
+            deadline = time.time() + self.FLUSH_INTERVAL
+
+    def run(self) -> None:
+        try:
```

Comment:
yeah, this is smarter. so i guess if this profiler crashed the sending thread hung waiting for the shutdown message?

Reply:
Yes, that was causing the hangs in our push e2e tests.